components/text-field/src/index.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Label from '@govuk-react/label';
+import Input from '@govuk-react/input';
+import { withWhiteSpace } from '@govuk-react/hoc';
+
+import InputMessaging from './atoms/input-messaging';
+
+/**
+ *
+ * ### Usage
+ *
+ * Simple
+ * ```jsx
+ * <TextField label="Label text" />
+ * ```
+ *
+ * With hint and an example error
+ * ```jsx
+ * <TextField label="Label text" hint="hint text" meta={{error: "some error information"}} />
+ * ```
+ *
+ * ### References:
+ * - https://github.com/alphagov/govuk-frontend/blob/master/src/components/input/_input.scss
+ * - https://github.com/alphagov/govuk_elements/blob/master/assets/sass/elements/_forms.scss
+ * - https://medium.com/@penx/form-elements-in-presentational-component-packages-a618e9aa7416
+ * - https://redux-form.com/7.1.2/docs/api/field.md/#input-props
+ * - https://redux-form.com/7.1.2/docs/api/field.md/#meta-props
+ */
+const TextField = ({
+  label, hint, input, meta, ...props
+}) => (
+  <Label error={meta.touched && meta.error} {...props}>
+    <InputMessaging label={label} hint={hint} meta={meta} />
```

Comment:
@stevesims I've swapped out `children` for `label` following the recent article linked. I think it's a good change in this case.